packages/apollo-server-fastify/src/ApolloServer.ts

```diff
@@ -24,134 +26,171 @@ export interface FastifyContext {
 
 export type ApolloServerFastifyConfig = Config<FastifyContext>;
 
+const pluginOptions: PluginOptions = {
+  fastify: '3 - 4',
+  name: 'apollo-server-fastify',
+};
+
 export class ApolloServer<
   ContextFunctionParams = FastifyContext,
 > extends ApolloServerBase<ContextFunctionParams> {
-  async createGraphQLServerOptions(
+
+  public plugin =
+    fp<ApolloFastifyPluginOptions>(
+      // Unbound this error if passed directly
+      (fastify, options) => this.#sharedHandler(fastify, options),
+      pluginOptions,
+    );
+
+  async #createGraphQLServerOptions(
     request: FastifyRequest,
     reply: FastifyReply,
   ): Promise<GraphQLOptions> {
     const contextParams: FastifyContext = { request, reply };
-    return this.graphQLServerOptions(contextParams);
+    return super.graphQLServerOptions(contextParams)
```

Comment:
I can't run tests with these changes, something about `super` in a private method (using `#`) is causing a syntax error. If I remove the `#` or revert back to `this` I don't see that error.

1) Can we just keep using `this`? Why the change to `super`?
2) I'm not opposed to using `#` as a private modifier, but the convention in the codebase is to use TS's `private`.